atef/config.py

```diff
@@ -179,8 +304,37 @@ def compare(self) -> Result:
                     f"for comparison {self.comparison}"
                 ),
             )
-        self.result = self.comparison.compare_signal(
-            self.signal,
+
+        try:
+            self.data = await self.get_data()
+        except TimeoutError:
+            return Result(
+                severity=self.comparison.if_disconnected,
+                reason=f"Signal disconnected when reading: {self.signal}"
+            )
+        except Exception as ex:
+            return Result(
+                severity=Severity.internal_error,
+                reason=(
+                    f"Getting data for signal {self.identifier!r} comparison "
+                    f"{self.comparison} raised {ex.__class__.__name__}: {ex}"
+                ),
+            )
+
+        if self.data is None:
+            # 'None' is likely incompatible with our comparisons and should
+            # be raised for separately
+            # TODO reviewers: Is this a sensible assumption?
```

Comment:
I cannot think of any situation where the user is checking a control system value and is expecting the Python `None` as a good result. Pyepics sometimes drops `None` in for disconnections, right?

Reply:
Yes, pyepics returns `None` which [ophyd interprets](https://github.com/bluesky/ophyd/blob/384d1a96c2a8956b5c3b3df62c7d72379485ba5a/ophyd/signal.py#L1216-L1220) as `TimeoutError` - so we shouldn't actually see it.

I think I'll remove this comment and leave in the check - unless we ever find a scenario where `None` could be meaningful in atef.